### PR TITLE
Invalid derivation of fixed fractionDigits

### LINKED ISSUE #34418: Invalid derivation of fixed fractionDigits facet fails to throw XmlSchemaException
Paragraph [4.3.1.12](https://www.w3.org/TR/xmlschema-2/#rf-fractionDigits) of the Xml Schema (version 1.0), part 2 specification states:

> If {fixed} is true, then types for which the current type is the {base type definition} cannot specify a value for fractionDigits other than {value}.

However, when a derived type overrides the fractionDigits facet with a different value, it is accepted by the XmlSchemaSet.Compile method.  An XmlSchemaException was expected.

Here is a unit test to demonstrate:
```
        [Fact]
        public void FractionDigitsFacetBaseFixed_Throws()
        {
            string schema = @"<?xml version='1.0' encoding='utf-8' ?>
<xs:schema elementFormDefault='qualified'
           xmlns:xs='http://www.w3.org/2001/XMLSchema'>
    <xs:simpleType name='foo'>
        <xs:restriction base='xs:decimal'>
            <xs:fractionDigits value='8' fixed='true'/>
        </xs:restriction>
    </xs:simpleType>
    <xs:simpleType name='bar'>
        <xs:restriction base='foo'>
            <xs:fractionDigits value='7'/>
        </xs:restriction>
    </xs:simpleType>
</xs:schema>
";
            XmlSchemaSet ss = new XmlSchemaSet();
            ss.Add(null, XmlReader.Create(new StringReader(schema)));

            Exception ex = Assert.Throws<XmlSchemaException>(() => ss.Compile());
            Assert.Contains("fractionDigits", ex.Message.ToLower());
            Assert.Contains("fixed", ex.Message.ToLower());

        }
```

### PR DESCRIPTION
**Pull request by mrj001**
@buyaa-n 
fix #34418

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/FacetChecker.cs [hunk 1]
```diff
@@ -330,6 +330,13 @@ internal void CompileFractionDigitsFacet(XmlSchemaFacet facet)
                 CheckDupFlag(facet, RestrictionFlags.FractionDigits, SR.Sch_DupFractionDigitsFacet);
                 _derivedRestriction.FractionDigits = XmlBaseConverter.DecimalToInt32((decimal)ParseFacetValue(_nonNegativeInt, facet, SR.Sch_FractionDigitsFacetInvalid, null, null));
 
+                if ((_baseFixedFlags & RestrictionFlags.FractionDigits) != 0)
```
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/src/System/Xml/Schema/FacetChecker.cs:333]: I think before this check better to check `if ((_derivedRestriction.FractionDigits != 0) && (_datatype.TypeCode != XmlTypeCode.Decimal))` and throw corresponding exception, please move

**Review by buyaa-n — APPROVED** [review]
@mrj001 Thank you for noticing this and fixing, left one NIT comment, other than that looks good to me

**Comment by mrj001** [comment]
@buyaa-n 
I've changed the order of these two tests.

Note that I also had to removed "fixed='true'" from another test (FractionDigitsMismatch_Throws).  It wasn't needed for this test, but at the time the test was created, it was not honored.   Therefore, I hadn't noticed the error.  With the fix to this issue, that test started failing due to a different exception being thrown now that the fixed facet was honored.

**Comment by buyaa-n** [comment]
> I've changed the order of these two tests.

> Note that I also had to removed "fixed='true'" from another test (FractionDigitsMismatch_Throws). It wasn't needed for this test, but at the time the test was created, it was not honored. Therefore, I hadn't noticed the error. With the fix to this issue, that test started failing due to a different exception being thrown now that the fixed facet was honored.

Thank you @mrj001, the tests might fail on full framework too, might need to add SkipOnTargetFramework

**Comment by buyaa-n** [comment]
> I've added the SkipOnTargetFramework attribute to FractionDigitsFacetBaseFixed_Throws, as .NET Framework 4.7.2 did not throw the required exception.

> .NET Framework 4.7.2 is also throwing the "totalDigits" error message in the FractionDigitsMismatch_Throws test instead of the "fractionDigits" error message.  This would cause that test to fail under 47.2 as well.  I've taken the liberty of adding the SkipOnTargetFramework attribute here as well, even though it is outside the scope of this pull request.

Great, looks good, thank you @mrj001.

The CI legs showing still in progress were successfully finished in details on Azure Pipelines, gonna merge this PR

**Pull request merged**